### PR TITLE
Core code for metadata uploads

### PR DESCRIPTION
**Pull request by JeffreyThiessen**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,55 @@
...
+
+#### CONFIG
+
+Specify a config file for connection to IRIDA. This is identical to an IRIDA Uploader config file.
+
+[Click here for more infomation about this file](https://irida-uploader.readthedocs.io/en/stable/configuration.html)
+
+#### METADATA_CSV
+
+Table must be in CSV format and must contain `sample` and `project_id` as the first 2 columns. All other columns will be processed as metadata.
```
Comment (DarianHole) [inline README.md]: These two columns could be found anywhere in the metadata file, just need to match the header strings of `sample` and `project_id`. I'd also mention that the project id has to be an integer value or the sample will be skipped over

File: upload_metadata.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        '-c',
+        '--config',
+        required=True,
+        help='IRIDA Uploader config file to parse'
+    )
+    parser.add_argument(
+        '-m',
+        '--metadata_csv',
+        required=True,
+        help='Metadata csv for upload containing matching ids to SampleList.csv in the FIRST column and a project id in the SECOND'
```
Comment (DarianHole) [inline upload_metadata.py]: Same thing as in the readme, the two columns just need to exist in the input file

File: README.md [hunk 3]
```diff
@@ -1 +1,55 @@
...
+SR21-0010f,3,MiSeq,2021-04-12,Cake
+CoV-20111,12,GridION,2021-06-09,Ice:Cream
+R3332-fad-34,23,MinION,2021-05-16,Cookies are great
+```
+
+##### Note: Any empty columns will be filled with `"NA"`
+
+#### --no_sample_creation (optional)
+
+Turns off sample creation for sample names not found in IRIDA
```
Comment (DarianHole) [inline README.md]: I'd just add in an output section as it does create a tracking file in the current directory. Something like:

```
### Output

#### Tracking `metadata_upload_status.csv`
Tracks the upload status of all the sample metadata in the upload

|    sample    | project_id | uploaded |    status    | 
|:------------:|:----------:|:---------------------:|:----------:|
|  SR21-0010f  |      3     |         True         |   |
|   CoV-20111  |     12     |        False        | Not in IRIDA and --no_sample_creation arg passed |
| R3332-fad-34 |     23a     |         False        | Unknown Project ID |
```

**Review by DarianHole — COMMENTED** [review]
Thanks for organizing the code Jeff! Just some small doc clarifications I think could be changed

File: README.md (hunk 3 above)
Comment (JeffreyThiessen) [reply README.md]: Fixed in https://github.com/phac-nml/irida-metadata-uploader/pull/1/commits/bee3cf0e15585992ab8e8436d29bf38f2895fc48

File: upload_metadata.py (hunk 2 above)
Comment (JeffreyThiessen) [reply upload_metadata.py]: Fixed in https://github.com/phac-nml/irida-metadata-uploader/pull/1/commits/c3e2baf188e14445c9a755235a1273664e7a9081

File: README.md (hunk 1 above)
Comment (JeffreyThiessen) [reply README.md]: Fixed in https://github.com/phac-nml/irida-metadata-uploader/pull/1/commits/c3e2baf188e14445c9a755235a1273664e7a9081

**Review by DarianHole — APPROVED** [review]
Thanks Jeff!

**Pull request merged**